_drafts/bazel-for-embedded.md

```diff
@@ -0,0 +1,389 @@
+
+## Target selection
+
+In addition selecting the correct toolchain, a platform can also be used to conditionally select targets during the build.
+For example, let’s imagine we have a library called `my_library` that relies on different backends depending on the architecture used.
+
+We can do the selection at build time as follow:
+
+```python
+cc_library(
+    name = “my_library“,
+    srcs = ["my_library.cc"],
+    deps = select({
+        ":arm_build": [":arm_lib"],
+        ":x86_build": [":x86_lib"],
+        "//conditions:default": [":generic_lib"],
+    }),
+)
+```
+
+The `select` built-in can be used for anything, such as compilation options, defines, dependencies, source files, etc. It lets you deal with variance at build time.
+Conditions can also be associative with the use of `selects.with_or` and `selects.config_setting_group` to achieve OR and AND boolean operations.
+
+## Target 
+
+Finally, when building all targets in the repository with a specific target platform, the incompatible ones will be skipped.
+
+Compatibility or incompatibility of a target is described with the `target_compatible_with` attribute of any rule. Note that this attribute propagates from its dependencies as well. In other words, if a target depends on a Windows library for example, and the latter is described with the appropriate `target_compatible_with` values, then if built on for a Linux platform, this all its parent target(s) will be skipped (not built).
+
+Here is for example how this could look like:
+
+```python
+cc_library(
+    name = "windows_lib",
+    srcs = ["windows_lib.cc"],
+    target_compatible_with = [
+        "@platforms//os:windows",
+    ],
+)
+```
+
+> Note that targets will be skipped only if all targets are requested to be built, using `bazel build //...` or `bazel test //...` for example. If a specific target is built and is incompatible, the build will fail.
+
+This is very useful when one wants to test all targets that are compatible with a specific platform either locally after a change or in CI for example.
+
+## Conclusion
+
+Bazel is a powerful and effective build tool for embedded systems, owing to its unique features and concepts. While this brief overview has provided an insight into its capabilities, there is much more to explore and uncover.
+
+However, Bazel has also a steep learning curve, which may pose a challenge for beginners. The effort put into mastering its main concepts is worthwhile, as it offers numerous advantages over traditional build systems such as Makefile and CMake.
+
+By leveraging Bazel's capabilities, developers can achieve faster build times, better reproducibility, and higher levels of code optimization, ultimately leading to more efficient and reliable embedded systems. Therefore, it is worth investing the time and resources necessary to gain a deep understanding of Bazel and incorporate it into your development workflow.
```

Comment:
Can you add a set of References for further reading? Would be helpful, especially if you know of some excellent public examples of `.bazelrc` files used for embedded.

Reply:
There are not much many bazel/embedded related resources out-there, at least as I know of but I will add some useful links for further reading.